app/uk/gov/hmrc/minorentityidentificationfrontend/services/ValidateTrustKnownFactsService.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.minorentityidentificationfrontend.services
+
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.minorentityidentificationfrontend.connectors.RetrieveTrustKnownFactsConnector
+import uk.gov.hmrc.minorentityidentificationfrontend.models.{DetailsMismatch, KnownFactsMatchingResult, SuccessfulMatch}
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class ValidateTrustKnownFactsService @Inject()(retrieveTrustKnownFactsConnector: RetrieveTrustKnownFactsConnector) {
+
+  def validateTrustKnownFacts(sautr: String, saPostcode: Option[String])(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[KnownFactsMatchingResult] = {
+    retrieveTrustKnownFactsConnector.retrieveTrustKnownFacts(sautr).map {
+      knownFacts =>
+        saPostcode match {
+          case None if knownFacts.isAbroad => SuccessfulMatch
+          case postcode if postcode == knownFacts.declarationPostcode | postcode == knownFacts.correspondencePostcode => SuccessfulMatch
+          case _ => DetailsMismatch
```

Comment:
If the post code is not defined and isAbroad is false should the result be NotEnoughInformationToMatch rather than  DetailsMismatch?

Reply:
I don't think so. We have enough information to get to this service but if postcode is not defined then isAbroad should be true so if it's not I was thinking it should be DetailsMismatch